experimental/zerotrie/src/reader.rs

```diff
@@ -367,6 +367,65 @@ pub fn get_bsearch_only(mut trie: &[u8], mut ascii: &[u8]) -> Option<usize> {
     }
 }
 
+/// Query the trie assuming all branch nodes are binary search
+/// and nodes use case-insensitive matching.
+pub fn get_ascii_bsearch_only_ignore_case(mut trie: &[u8], mut ascii: &[u8]) -> Option<usize> {
+    loop {
+        let (b, x, i, search);
+        (b, trie) = trie.split_first()?;
+        let byte_type = byte_type(*b);
+        (x, trie) = match byte_type {
+            NodeType::Ascii => (0, trie),
+            NodeType::Span => {
+                debug_assert!(false, "Span node found in ASCII trie!");
+                return None;
+            }
+            NodeType::Value => read_varint_meta3(*b, trie),
+            NodeType::Branch => read_varint_meta2(*b, trie),
+        };
+        if let Some((c, temp)) = ascii.split_first() {
+            if matches!(byte_type, NodeType::Ascii) {
+                if b.to_ascii_lowercase() == c.to_ascii_lowercase() {
```

Comment:
issue: this line and the bsearch line are the only differences between the previous, right? can this be refactored to be one generic function that takes in a search key closure?

(keep it specialized for the public api)

Reply:
This file contains several versions of the same function independently because this is the hot path and I'd rather it be a standalone function. The function it not itself very long so it's easy to copy, paste, and modify. I can try to merge the impls but it's not clear it would make the code more readable.

Reply:
If the function were generic and took a closure it should monomorphise. Happy to be convinced otherwise on readability though, right now this and the above function seem mergeable, I'm not sure of the others